### PR TITLE
bug: Disable flunky `test_build_receipt_hashes_with_num_shard` check, but keep the test

### PR DESCRIPTION
**Pull request by pmnoxx**
Let's disable `test_build_receipt_hashes` test. This test is based on timing assumption that code `A` should be faster than `B`. That's not true while compiling tests with `-O2`. Let's disable tor remove this test.

This test always fails, when running with `-O2`.
```
L .~/r/nearcore.master piotr-add-tests-to-flunky-test NO_REMOTE cargo test --tests test_build_receipt_hashes
```

```
---- test_utils::test::test_build_receipt_hashes stdout ----
thread 'test_utils::test::test_build_receipt_hashes' panicked at 'naive duration vs production 7.855917ms 6.452102ms', chain/chain/src/test_utils.rs:1471:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Closes https://github.com/near/nearcore/issues/5836

### CODE REVIEW

File: chain/chain/src/test_utils.rs [hunk 1]
```diff
@@ -1468,10 +1468,16 @@ mod test {
...
+        assert!(
+            2 * naive_duration > 3 * prod_duration,
+            "naive duration vs production {:?} {:?}",
+            naive_duration,
+            prod_duration
+        );
     }
 
     #[test]
+    #[ignore]
```
Comment (mm-near) [inline chain/chain/src/test_utils.rs:1480]: add a comment with github issue that explains more info on why we ignored this test.

**Review by mm-near — APPROVED** [review]

File: chain/chain/src/test_utils.rs (hunk 1 above)
Comment (pmnoxx) [reply chain/chain/src/test_utils.rs:1480]: https://github.com/near/nearcore/issues/5836

**Review by bowenwang1996 — COMMENTED** [review]
I suggest that we also figure out either how to fix the test or remove it altogether

**Comment by pmnoxx** [comment]
> I suggest that we also figure out either how to fix the test or remove it altogether

First of all, what is the purpose of this test? `cargo test` runs code without optimizations by default. I would say, writing any test, which benchmarks performance is pointless, as you are not really testing the real code. And it should be removed.

**Comment by mzhangmzz** [comment]
> > I suggest that we also figure out either how to fix the test or remove it altogether
> 
> First of all, what is the purpose of this test? `cargo test` runs code without optimizations by default. I would say, writing any test, which benchmarks performance is pointless, as you are not really bencharking the real code. I think it should be removed.

I think this test also tests that build_receipts_hashes and naive_build_receipts_hashes return the same result, which is not about performance, but correctness of the code. I'm ok with removing the timing line, but I think we should keep it the test there

**Comment by pmnoxx** [comment]
> > > I suggest that we also figure out either how to fix the test or remove it altogether
> > 
> > 
> > First of all, what is the purpose of this test? `cargo test` runs code without optimizations by default. I would say, writing any test, which benchmarks performance is pointless, as you are not really bencharking the real code. I think it should be removed.
> 
> I think this test also tests that build_receipts_hashes and naive_build_receipts_hashes return the same result, which is not about performance, but correctness of the code. I'm ok with removing the timing line, but I think we should keep it the test there

@mzhangmzz I absolute agree. This test is good, it's a regression tests, which makes sure we don't break anything in the future. Let's remove just the line about the timing, and close the issue.

**Comment by bowenwang1996** [comment]
Sounds good. Let's remove the timing check then

**Pull request merged**